### ISSUE TITLE
Mouse selecting a domain where there are no datapoints, crashes the tab

### ISSUE THREAD

**Issue by nonvex** [leeoniya/uPlot#640]
Example demo page: https://leeoniya.github.io/uPlot/demos/nice-scale.html

Zoom in a domain where there are no more data points:
![Screenshot 2021-12-27 at 20 05 56](https://user-images.githubusercontent.com/77842060/147500312-b375e052-0372-4619-9617-9364b81a7f53.png)

--
_PS: Many thanks for uPlot. I am a happy user._

**Comment by leeoniya** [comment]
yeah, it's a bug in the demo, which s broken in multiple other ways.

this demo relies on a double re-render to work around the independence of scale and axis configs, which i've started to realize cannot really be decoupled for a number of additional reasons. i'm hoping this will be addressed by tighter integration between axes and scales (https://github.com/leeoniya/uPlot/tree/ticks-axis-incrs), but doing it in a way that doesnt have measurable streaming performance slowdowns is proving to be tricky.
